service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +74,279 @@ public void disbandTeam(JWTData userData, int teamId) {
+                    usersTeams.setTeamRole(TeamRole.PENDING);
+                    usersTeams.store();
+                    // TODO SEND EMAIL TO ACCEPT ROLE ON TEAM
+                  } else {
+                    // TODO SEND EMAIL WITH INVITE TO CREATE USER / REQUEST TO JOIN TEAM
+                  }
+                }
+              });
+    } else {
+      throw new WrongTeamRoleException(
+          inviteUserRequest.getTeamId(), inviterUserTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public List<UsersTeamDataResponse> getApplicants(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    List<UsersTeamsRecord> applicants =
+        db.selectFrom(USERS_TEAMS).where(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.PENDING)).fetch();
+    List<UsersTeamDataResponse> responses = new ArrayList<>();
+    applicants.forEach(
+        app -> {
+          UsersTeamDataResponse response =
+              new UsersTeamDataResponse(app.getUserId(), teamId, app.getTeamRole());
+          responses.add(response);
+        });
+    return responses;
+  }
+
+  @Override
+  public void applyToTeam(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    if (db.fetchExists(
+        USERS_TEAMS.where(
+            USERS_TEAMS
+                .TEAM_ID
+                .eq(teamId)
+                .and(USERS_TEAMS.USER_ID.eq(userData.getUserId()))))) {
+      throw new MemberApplicationException(
+          teamId, userData.getUserId());
+    } else {
+      UsersRecord invitedUser =
+          db.selectFrom(USERS).where(USERS.ID.eq(userData.getUserId())).fetchOne();
+      if (invitedUser != null) {
+        UsersTeamsRecord usersTeams = db.newRecord(USERS_TEAMS);
+        usersTeams.setTeamId(teamId);
+        usersTeams.setUserId(invitedUser.getId());
+        usersTeams.setTeamRole(TeamRole.PENDING);
+        usersTeams.store();
+      } else {
+        throw new UserDoesNotExistException(userData.getUserId());
+      }
+    }
+  }
+
+  @Override
+  public void approveUser(JWTData userData, int teamId, int memberId) {
+    checkUserExists(memberId);
+    checkTeamExists(teamId);
+    UsersTeamsRecord leaderTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+    if (leaderTeamsRecord != null && leaderTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      UsersTeamsRecord approvedUserRecord =
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.USER_ID.eq(memberId)))
+              .fetchOne();
+      if (approvedUserRecord.getTeamRole() == TeamRole.PENDING) {
+        approvedUserRecord.setTeamRole(TeamRole.MEMBER);
+        approvedUserRecord.update();
+      } else {
+        throw new MemberStatusException(
+            memberId, teamId);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          leaderTeamsRecord.getTeamId(), leaderTeamsRecord.getTeamRole());
```

Comment:
WrongTeamRoleException takes the role that the user is **supposed to** have not what they do have. Anywhere where you call WrongTeamRoleException make sure you're passing in a static role instead of one from the user